auth-tokens/src/main/java/com/palantir/tokens/auth/UnverifiedJsonWebToken.java

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright 2016 Palantir Technologies, Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.tokens.auth;
+
+import static com.google.common.base.Preconditions.checkArgument;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.datatype.guava.GuavaModule;
+import com.google.common.io.BaseEncoding;
+import java.io.IOException;
+import java.nio.ByteBuffer;
+import java.util.UUID;
+import org.immutables.value.Value;
+
+/**
+ * Represents the parsed form of a JWT but does NOT verify the token signature and
+ * content must not be used for any security purposes.
+ * <p>
+ * This class provides informational (at best) data from a JWT, primarily for logging
+ * purposes.
+ */
+@Value.Immutable
+@Value.Style(visibility = Value.Style.ImplementationVisibility.PACKAGE, jdkOnly = true)
+public abstract class UnverifiedJsonWebToken {
+
+    private static final ObjectMapper MAPPER = new ObjectMapper()
+            .registerModule(new GuavaModule());
+
+    /**
+     * Returns an unverified user id.
+     */
+    @Value.Parameter
+    public abstract String getUnverifiedUserId();
+
+    /**
+     * Attempts to create an {@link UnverifiedJsonWebToken} from provided {@link BearerToken}.
+     * <p>
+     * Note: this method will NOT perform signature verification and thus the content must not
+     * be used for any security purposes.
+     */
+    public static UnverifiedJsonWebToken of(BearerToken token) {
+        String[] segments = token.getToken().split("\\.");
+        checkArgument(segments.length == 3, "Invalid JWT: expected 3 segments, found %s", segments.length);
+
+        JsonWebTokenPayload payload = extractPayload(segments[1]);
+
+        return ImmutableUnverifiedJsonWebToken.of(decodeUuidBytes(payload.getSub()));
+    }
+
+    private static JsonWebTokenPayload extractPayload(String payload) {
+        try {
+            return MAPPER.readValue(BaseEncoding.base64().decode(payload), JsonWebTokenPayload.class);
+        } catch (IOException e) {
+            throw new IllegalArgumentException("Invalid JWT: cannot parse payload", e);
+        }
+    }
+
+    private static String decodeUuidBytes(byte[] bytes) {
+        checkArgument(bytes.length == 16, "Invalid JWT: cannot decode UUID, require 16 bytes, found %s", bytes.length);
```

Comment:
true. let's leave it as is.

On Wed, Jun 8, 2016 at 4:54 PM Mark Elliot <notifications@github.com> wrote:

> In
> auth-tokens/src/main/java/com/palantir/tokens/auth/UnverifiedJsonWebToken.java
> <https://github.com/palantir/auth-tokens/pull/10#discussion_r66271422>:
>
> > +
> > +        JsonWebTokenPayload payload = extractPayload(segments[1]);
> > +
> > +        return ImmutableUnverifiedJsonWebToken.of(decodeUuidBytes(payload.getSub()));
> > +    }
> > +
> > +    private static JsonWebTokenPayload extractPayload(String payload) {
> > +        try {
> > +            return MAPPER.readValue(BaseEncoding.base64().decode(payload), JsonWebTokenPayload.class);
> > +        } catch (IOException e) {
> > +            throw new IllegalArgumentException("Invalid JWT: cannot parse payload", e);
> > +        }
> > +    }
> > +
> > +    private static String decodeUuidBytes(byte[] bytes) {
> > +        checkArgument(bytes.length == 16, "Invalid JWT: cannot decode UUID, require 16 bytes, found %s", bytes.length);
>
> Might be a bit weird if someone has a 16 byte string for sub/there's no
> easy way to distinguish things that are valid/invalid UUIDs.
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/palantir/auth-tokens/pull/10/files/6d4d17ad4debef8a88d281999ae757b082ad95ad#r66271422>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AS4-yra9HgKYmwbYf_yZ-vgdRdBFLeRWks5qJteggaJpZM4IwnFn>
> .
>